api/types/types.go

```diff
@@ -600,3 +600,10 @@ type BuildCache struct {
 	Parent      string
 	Description string
 }
+
+// BuildCachePruneOptions hold parameters to prune the build cache
+type BuildCachePruneOptions struct {
+	All          bool
+	KeepDuration time.Duration
```

Comment:
Gonna summarize a conversation that @tiborvass and I had offline. If we've understood the criteria for pruning correctly a given cache layer will be kept if it meets the `KeepDuration` criterion and meets the `KeepStorage` criterion. In other words keep rules are conjunctive and by De Morgan's law, purge rules are disjunctive 😜 

If that's the case I would rename `KeepDuration` to `KeepMaxAge`. Duration implies that the cached item will be kept around for that amount of time, whereas max age implies it will be kept for a maximum of that amount of time given all other criteria are also met.